CoreFoundation/String.subproj/CFString.c

```diff
@@ -4117,7 +4117,7 @@ CFStringRef CFStringCreateByCombiningStrings(CFAllocatorRef alloc, CFArrayRef ar
             } else {
                 if (!isSepCFString) { // NSString
                     CFStringGetCharacters(separatorString, CFRangeMake(0, CFStringGetLength(separatorString)), (UniChar *)bufPtr);
-                } else if (canBeEightbit) {
+                } else if (canBeEightbit || __CFStrIsUnicode(separatorString)) {
```

Comment:
only for objc, but not swift; which is a different bug.
I wonder if the perf differential is worth it for just avoiding the isSepCFString = false in the case of NSStrings in general by instead claiming it as "can be eight bit" by just calling CFStrIsUnicode instead of testing if it is a NSString (definitely merits further investigation)